presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -108,8 +112,9 @@ public Rewriter(Metadata metadata, SqlParser sqlParser, Session session, Map<Sym
         public PlanNode visitProject(ProjectNode node, RewriteContext<Void> context)
         {
             PlanNode source = context.rewrite(node.getSource());
-            Map<Symbol, Expression> assignments = ImmutableMap.copyOf(Maps.transformValues(node.getAssignments(), this::simplifyExpression));
-            return new ProjectNode(node.getId(), source, assignments);
+            expressionAssignments = node.getAssignments();
```

Comment:
Hmmmm. I need to pass the assignments of the `ProjectNode` that I'm currently visiting as context to the rewriter. I can't pass additional context to `simplifyExpression` where the rewriters are called because it has a set function definition, which is why I put it in a class field. I'm not sure of a better way to get the context to the rewriter.